x-pack/plugins/index_lifecycle_management/public/store/actions/nodes.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createAction } from 'redux-actions';
+import { toastNotifications } from 'ui/notify';
+import { loadNodes, loadNodeDetails } from '../../services/api';
+import { SET_SELECTED_NODE_ATTRS } from '../constants';
+
+export const setSelectedNodeAttrs = createAction(SET_SELECTED_NODE_ATTRS);
+export const setSelectedPrimaryShardCount = createAction(
+  'SET_SELECTED_PRIMARY_SHARED_COUNT'
+);
+export const setSelectedReplicaCount = createAction(
+  'SET_SELECTED_REPLICA_COUNT'
+);
+export const fetchedNodes = createAction('FETCHED_NODES');
+let fetchingNodes = false;
+export const fetchNodes = () => async dispatch => {
+  try {
+    if (!fetchingNodes) {
+      fetchingNodes = true;
+      const nodes = await loadNodes();
+      dispatch(fetchedNodes(nodes));
+    }
+  } catch (err) {
+    return toastNotifications.addDanger(err.data.message);
+  } finally {
+    fetchingNodes = false;
+  }
+};
+
+export const fetchedNodeDetails = createAction(
+  'FETCHED_NODE_DETAILS',
+  (selectedNodeAttrs, details) => ({
```

Comment:
Personal nit: I find the `createAction` interface opaque and difficult to work with. With vanilla actions (and `redux-thunk`) I feel like I'm more in control of the code. I don't think we need to address this in this PR, just wanted to share my perspective.

Reply:
I like createAction personally.